Mindbox.Data.Linq/Mapping/AttributedMetaModel.cs

```diff
@@ -92,46 +90,15 @@ public override IEnumerable<MetaTable> GetTables()
 	        if (staticTables.Count > 0)
 		        return staticTables;
 
-	        metaModelLock.AcquireReaderLock(Timeout.Infinite);
-	        try 
-			{
-		        return metaTables.Values.Where(metaTable => metaTable != null).Distinct();
-	        }
-	        finally 
-			{
-		        metaModelLock.ReleaseReaderLock();
-	        }
+			return metaTables.Values.Where(metaTable => metaTable != null).Distinct();
```

Comment:
Enumeration is not thread-safe

Reply:
The only way I see to fix this is to wrap every access to `metaTables` in `ReaderWriterLock`, like it was before I`d deleted it :-)
But https://msdn.microsoft.com/en-us/library/dd287131.aspx says that you wouldn't get at least any exceptional situation while using ConcurrentDictionary.GetEnumerator in concurrent environment. You woudn't get a snapshot of values, but Enumerator will work correctly, it will return every registered metatable, except in some cases metatables, that are currently adding to the dictionary. Or maybe you think it wouldn't return every registered values because of internal structure of the ConcurrentDictionary?

Reply:
[This gist](https://gist.github.com/timramone/f347e756a4d63f32e8d3b6923743a539) shows my point. After reading from the enumerator half of its entiries, I add to the concurrent dictionary some other entries, and then continue reading from that enumerator. In the end I got all the entries that was in the concurrent dictionary at the moment I got enumerator, plus some entries that has been added to the dictionary afterwards.

Reply:
You're right. It is thread-safe. It just may have some future changes in it. As long as we only have single item inserts and don't have to ensure consistency between different items, that should not cause any problems.